pkg/daemon/ceph/osd/volume.go

```diff
@@ -285,14 +287,20 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 			}...)
 		}
 		mdArgs = append(mdArgs, strings.Split(conf["devices"], " ")...)
+		r, err := regexp.Compile("([a-z0-9]+)")
+		if err != nil {
+			return fmt.Errorf("failed to compile regex. %+v", err)
+		}
+		mdList := strings.Split(strings.Replace(r.ReplaceAllString(md, "/dev/$1"), " ", "", -1), ",")
```

Comment:
So this will not support devices which are specified by `/dev/disk/by-...`. @travisn isn't this supposed to be supported for OSD disks, and shouldn't we support this for metadata disks as well?